lib/Service/ParticipantService.php

```diff
@@ -471,6 +482,49 @@ public function markUsersAsMentioned(Room $room, array $userIds, int $messageId)
 		$query->execute();
 	}
 
+	public function updateReadPrivacyForActor(string $actorType, string $actorId, int $readPrivacy): void {
+		$query = $this->connection->getQueryBuilder();
+		$query->update('talk_attendees')
+			->set('read_privacy', $query->createNamedParameter($readPrivacy, IQueryBuilder::PARAM_INT))
+			->where($query->expr()->eq('actor_type', $query->createNamedParameter($actorType)))
+			->andWhere($query->expr()->eq('actor_id', $query->createNamedParameter($actorId)));
+		$query->execute();
+	}
+
+	public function getLastCommonReadChatMessage(Room $room): int {
+		$query = $this->connection->getQueryBuilder();
+		$query->selectAlias($query->func()->min('last_read_message'), 'last_common_read_message')
+			->from('talk_attendees')
+			->where($query->expr()->eq('actor_type', $query->createNamedParameter(Attendee::ACTOR_USERS)))
+			->andWhere($query->expr()->eq('room_id', $query->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)))
+			->andWhere($query->expr()->eq('read_privacy', $query->createNamedParameter(Participant::PRIVACY_PUBLIC, IQueryBuilder::PARAM_INT)));
+
+		$result = $query->execute();
+		$row = $result->fetch();
+		$result->closeCursor();
+
+		return (int) ($row['last_common_read_message'] ?? 0);
+	}
+
+	public function getLastCommonReadChatMessageForMultipleRooms(array $roomIds): array {
```

Comment:
PHPDoc would be nice, especially when the return value is not a plain indexed array

Reply:
it's a map of roomid => messageid, but sure will add